tools/donate_cpu_lib.py

```diff
@@ -81,28 +99,18 @@ def get_cppcheck_info(cppcheck_path):
         return ''
 
 
-def compile_version(work_path, jobs, version):
-    if os.path.isfile(work_path + '/' + version + '/cppcheck'):
+def compile_version(cppcheck_path, jobs):
+    if os.path.isfile(os.path.join(cppcheck_path, 'cppcheck')):
         return True
-    os.chdir(work_path + '/cppcheck')
-    subprocess.call(['git', 'checkout', version])
-    subprocess.call(['make', 'clean'])
-    subprocess.call(['make', jobs, 'MATCHCOMPILER=yes', 'CXXFLAGS=-O2 -g'])
-    if os.path.isfile(work_path + '/cppcheck/cppcheck'):
-        os.mkdir(work_path + '/' + version)
-        dest_path = work_path + '/' + version + '/'
-        subprocess.call(['cp', '-R', work_path + '/cppcheck/cfg', dest_path])
-        subprocess.call(['cp', 'cppcheck', dest_path])
-    subprocess.call(['git', 'checkout', 'main'])
-    try:
-        subprocess.call([work_path + '/' + version + '/cppcheck', '--version'])
-    except OSError:
-        return False
-    return True
+    # Build
+    ret = compile_cppcheck(cppcheck_path, jobs)
+    # Clean intermediate build files
+    os.chdir(cppcheck_path)
+    subprocess.call(['git', 'clean', '-f', '-d', '-x', '--exclude', 'cppcheck'])
+    return ret
 
 
 def compile_cppcheck(cppcheck_path, jobs):
-    print('Compiling Cppcheck..')
```

Comment:
This is not dropped by moved outside of the function, into to donate-cpu.py as shown below:

print('Compiling Cppcheck-{}..'.format(ver))

Reply:
Ah - I missed that.

The library code is also used by `test-my-pr.py` - so that probably needs adjusting too :/ There's a bit too much logic in the higher-level scripts which is not good.

Reply:
Okay, I'll undo this move.